### ISSUE TITLE
RS1024 reported for explicit reference equals

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#2335]
#### Analyzer package

[Microsoft.CodeAnalysis.Analyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers)

#### Package Version

3.0.0-beta2.19213.1+14035759 (built from 14035759)

#### Diagnostic ID

RS1024 (Compare symbols correctly)

#### Repro steps

```csharp
ISymbol x;
ISymbol y;
bool same = (object)x == y;
```

#### Expected behavior

Unknown

#### Actual behavior

RS1024 is reported.
